src/mapdata.h

```diff
@@ -168,59 +168,71 @@ void init_ter_bitflags_map();
- std::set<std::string> flags;// string flags which may or may not refer to what's documented above.
- unsigned long bitflags; // bitfield of -certian- string flags which are heavily checked
- iexamine_function examine; //What happens when the terrain is examined
- std::string harvestable; //what will be harvested from this terrain?
- std::string transforms_into; // transform into what terrain?
- int harvest_season; // when will this terrain get harvested?
- int bloom_season; // when does this terrain bloom?
- std::string open; //open action: transform into terrain with matching id
- std::string close; //close action: transform into terrain with matching id
-
- map_bash_info bash;
- map_deconstruct_info deconstruct;
- // Maximal volume of items that can be stored in/on this furniture
- int max_volume;
-
- bool has_flag(const std::string & flag) const {
-     return flags.count(flag) != 0;
- }
+struct map_data_common_t {
+    std::string id;    // The terrain's ID. Must be set, must be unique.
+    std::string name;  // The plaintext name of the terrain type the user would see (i.e. dirt)
+    std::string open;  // Open action: transform into terrain with matching id
+    std::string close; // Close action: transform into terrain with matching id
 
- bool has_flag(const ter_bitflags flag) const {
-     return (bitflags & mfb(flag));
- }
+    map_bash_info        bash;
+    map_deconstruct_info deconstruct;
 
- void set_flag(std::string flag) {
-     flags.insert(flag);
-     if("TRANSPARENT" == flag) {
-         transparent = true;
-     }
-     if ( ter_bitflags_map.find( flag ) != ter_bitflags_map.end() ) {
-         bitflags |= mfb ( ter_bitflags_map.find( flag )->second );
-     }
- }
- bool transparent;
+    std::set<std::string> flags;    // string flags which possibly refer to what's documented above.
+    unsigned long         bitflags; // bitfield of -certian- string flags which are heavily checked
+
+    /*
+    * The symbol drawn on the screen for the terrain. Please note that there are extensive rules
+    * as to which possible object/field/entity in a single square gets drawn and that some symbols
+    * are "reserved" such as * and % to do programmatic behavior.
+    */
+    long sym;
+
+    int loadid;     // This is akin to the old ter_id, however it is set at runtime.
+    int movecost;   // The amount of movement points required to pass this terrain by default.
+    int max_volume; // Maximal volume of items that can be stored in/on this furniture
+    
+    nc_color color; //The color the sym will draw in on the GUI.
+
+    iexamine_function examine; //What happens when the terrain is examined
+
+    bool transparent;
+
+    bool has_flag(const std::string & flag) const {
+        return !!flags.count(flag);
+    }
+
+    bool has_flag(const ter_bitflags flag) const {
+        return (bitflags & mfb(flag));
+    }
+
+    void set_flag(std::string flag) {
+        flags.insert(std::move(flag));
```

Comment:
What do you mean here? Take a careful look at what is happening here it one more time (it probably doesn't help that the names are flags and flag).